Added Michael Newton's files to Michael's branch.

cogerk opened the pull request:
This code was written by Michael Newton and is intended to use Bokah to plot some sample data.

cogerk committed:
Added Michael Newton's files to Michael's branch.

cogerk commented on Project/dataplot.py:
```diff
@@ -0,0 +1,33 @@
+
+# Plot a Specified Column in Dataframe
+lineplot = Line(sampleSI['DO'], xlabel='Date', ylabel='DO', legend="top_left")
+checkbox_button_group = CheckboxButtonGroup(labels=["DO", "NH4", "pH"], active=[0, 1])
```
It looks like NH4 and pH buttons don't have functionality so we should remove those labels.

manewton replied:
Actually none of the buttons have functionality at this point. I do anticipate imparting functionality to these buttons to easily switch between some relevant data.

cogerk commented on Project/dataplot.py:
```diff
@@ -0,0 +1,33 @@
+checkbox_button_group = CheckboxButtonGroup(labels=["DO", "NH4", "pH"], active=[0, 1])
+
+# This callback is apparently only usable for a slider
+# callback = CustomJS(args=dict(source=source), code="""
```
What's a callback?

dacb replied:
A callback is a function that gets called when the user interacts with the slider.  In this case, the callback is a javascript function.

cogerk replied:
Dave you're so fast!!  Thank you.  Can't run javascript in a python environment right? is this why it's commented out?

dacb replied:
That javascript is run client side in the web browser.  It is loaded by the browser as part of the page.  I'm not sure why it is commented out, in this instance.

manewton replied:
I commented out the callback because I initially added it thinking that it could be used to impart functionality to the buttons... but the buttons widget doesn't accept callbacks... I've kept it there so that if I want to add a slider I can easily do so in the future.

cogerk commented on Project/dataplot.py:
```diff
@@ -0,0 +1,33 @@
+import numpy as np
```
Unused imports = CustomJS & output_notebook(in commented out block), should we remove?  Ok if no, just curious on reasoning

cogerk replied:
Recommend  adding a block of text in the beginning to describe functionality, something like...
"""
Written By: a name
For: Winkler Lab/CSE599 Winter Quarter 2016
Purpose: a purpose

To Do:
+A couple
+of ideas
+of where to go next
"""

manewton replied:
I will add more description on functionality and things to do which should address a number of concerns.

I do have unused imports but I anticipate using them in the future as I expand the plotting and server options.

cogerk commented on Project/dataplot.py:
```diff
@@ -0,0 +1,33 @@
+from bokeh.charts import *
+from bokeh.io import output_notebook, show, output_file, vform
+from bokeh.models import ColumnDataSource, CustomJS
+import pandas as pd
```
this import should live next to his buddy numpy, recommend making this line 2

cogerk commented on Project/dataplot.py:
```diff
@@ -0,0 +1,33 @@
+
+# output_notebook()
+output_file("doplot.html", title="DO Line Plot")
+show(vform(checkbox_button_group, lineplot))
```
Yay!  Bokeh is so cool!  Thank you for putting this together for us Michael!

cogerk commented on Project/dataplot.py:
```diff
@@ -0,0 +1,33 @@
+#         source.trigger('change');
+# """)
+
+# output_notebook()
```
Why is this commented out?

manewton committed:
Update dataplot.py

Removed unused import statements. Added a description of its purpose at the head of code. Callback remained commented to allow for easy addition of slider callback in the future if needed.
